js/main.js

```diff
@@ -87,55 +110,106 @@ var renderMapPin = function (template, ad) {
+  return Math.round(сalculatedX) + ', ' + Math.round(сalculatedY);
 };
 
-var renderPopup = function (template, ad) {
-  var popupElement = template.cloneNode(true);
-  var popupElementFeatures = popupElement.querySelector('.popup__features').children;
-  var popupElementPhoto = popupElement.querySelector('.popup__photo');
-
-  popupElement.querySelector('.popup__avatar').src = ad.author.avatar;
-  popupElement.querySelector('.popup__title').textContent = ad.offer.title;
-  popupElement.querySelector('.popup__text--address').textContent = ad.offer.address;
-  popupElement.querySelector('.popup__text--price').textContent = ad.offer.price + ' ₽/ночь';
-  popupElement.querySelector('.popup__type').textContent = TYPES_MAP[ad.offer.type];
-  popupElement.querySelector('.popup__text--time').textContent = 'Заезд после ' + ad.offer.checkin + ', выезд до ' + ad.offer.checkout;
-  popupElement.querySelector('.popup__description').textContent = ad.offer.description;
-
-  popupElement.querySelector('.popup__text--capacity').textContent = ad.offer.rooms +
-    getDeclOfNumb(ad.offer.rooms, [' комната', ' комнаты', ' комнат']) + ' для ' +
-    ad.offer.guests + getDeclOfNumb(ad.offer.guests, [' гостя', ' гостей', ' гостей']);
-
-  for (var i = 0; i < popupElementFeatures.length - ad.offer.features.length; i++) {
-    popupElementFeatures[popupElementFeatures.length - 1 - i].style.display = 'none';
+var toggleDisabledOnFormNodes = function (isDisable) {
+  var pointerEventsValue = isDisable === true ? 'none' : 'auto';
+
+  for (var i = 0; i < formFieldsNodes.length; i++) {
+    formFieldsNodes[i].disabled = isDisable;
+    formFieldsNodes[i].style.pointerEvents = pointerEventsValue;
+  }
+
+  for (i = 0; i < mapFiltersNodes.length; i++) {
+    mapFiltersNodes[i].disabled = isDisable;
+    mapFiltersNodes[i].style.pointerEvents = pointerEventsValue;
   }
+};
+
+var unlockPage = function (evt) {
+  if (evt.button === 0 || evt.key === 'Enter') {
+    for (var i = 0; i < adObjectsArr.length; i++) {
+      var adObjectsArrItem = adObjectsArr[i];
 
-  popupElementPhoto.src = ad.offer.photos[0];
-  if (ad.offer.photos.length > 1) {
-    for (i = 1; i < ad.offer.photos.length; i++) {
-      popupElement.querySelector('.popup__photos').appendChild(popupElementPhoto.cloneNode(true)).src = ad.offer.photos[i];
+      fragment.appendChild(renderMapPin(mapPinTemplateNode, adObjectsArrItem));
     }
+    mapPinsNode.appendChild(fragment);
+
+    toggleDisabledOnFormNodes(false);
+    mapNode.classList.remove('map--faded');
+    formNode.classList.remove('ad-form--disabled');
+
+    addressInputNode.value = getAddressMapPinMainStr(false);
+
+    mapPinMainNode.removeEventListener('mousedown', unlockPage);
+    mapPinMainNode.removeEventListener('keydown', unlockPage);
   }
+};
+
+var validateGuestsSelect = function () {
+  var validMessage = ROOMS_FOR_GUESTS_MAP[roomsSelectNode.value].includes(guestsSelectNode.value) === true ? '' : 'Такой вариант не подходит!';
```

Comment:
Давай сделаем более описательно
```suggestion
  var customValidityMessage = ROOMS_FOR_GUESTS_MAP[roomsSelectNode.value].includes(guestsSelectNode.value) === true ? '' : 'Такой вариант не подходит!';
```